### DISCUSSION TITLE
Large difference of results (homework 3)

### DISCUSSION THREAD

**Discussion by Di3goCH** [cafawo/FinancialDataAnalytics#6] [Q&A] [answered]
Hi all, 

I am not sure if my results are ok, but I got a large difference between the methods we have to use (Numerical Integration and Monte Carlo Simulation) when I calculated the option's price. I got around 5 and 90 respectively. 

 Someone else got large differences? I am not sure if it could happen, or if the results should be more similar between them.

Thanks in advance

**Comment by birte0** [comment]
Typically, the results of monte carlo and numnerical integration for calculating the value of a call option should be similar (up to a small simulation error). Ideally, it would be helpful if you could share some insights into how you computed the call option value using each method.

**Reply by Di3goCH** [reply]
Hi, I have doubts about the numerical integration. x is the price at maturity and K the strike price

I calculate the integral of **(x-K) * lognormal function** with the parameters given in the exercise. 

I use **integrate.squad()** to integrate the function from K to +infinite, because for **x<K** the buyer does not use the call option, since he can buy the stock for a price **x** that is lower than **K**, and finally I multiply the result of the integrations for the discount factor. Should I share the code here?

**Reply by cafawo** [reply, marked as answer]
I integrate over the Log-Normal PDF using `from scipy.integrate import quad`.

PS: 90 sounds better than 5.

**Reply by Di3goCH** [reply]
Thanks! I used that function for integration before, but the problem was with the function to integrate. I tried to use lognormal.pdf() from stats, but it didn't work because of something with the parameters. I used **stast.lognormal.pdf(x,sigma,mu)**, anyway the results was much smaller.  Finally, I got similar results to that from the Monte Carlo simulation (around 90) only by writing the whole function given in the homework.
